src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -330,7 +337,7 @@ private async ValueTask<Http3ControlStream> CreateNewUnidirectionalStreamAsync<T
                 RemoteEndPoint = streamContext.RemoteEndPoint as IPEndPoint
             };
 
-            return new Http3ControlStream<TContext>(application, this, httpConnectionContext);
+            return new Http3ControlStream<TContext>(application, this, httpConnectionContext, _serverSettings);
```

Comment:
With HTTP/2 the settings are on the stream context. Add server settings to Http3StreamContext?